worldedit-cli/src/main/java/com/sk89q/worldedit/cli/data/FileRegistries.java

```diff
@@ -45,36 +48,28 @@ public FileRegistries(CLIWorldEdit app) {
     }
 
     public void loadDataFiles() {
-        ResourceLoader resourceLoader = WorldEdit.getInstance().getPlatformManager().queryCapability(Capability.CONFIGURATION).getResourceLoader();
-        try {
-            URL url = resourceLoader.getResource(FileRegistries.class, app.getPlatform().getDataVersion() + ".json");
-            this.dataFile = gson.fromJson(Resources.toString(url, StandardCharsets.UTF_8), DataFile.class);
+        Path outputFolder = WorldEdit.getInstance().getWorkingDirectoryPath("cli-data");
+        Path checkPath = outputFolder.resolve(app.getPlatform().getDataVersion() + "_" + CLI_DATA_VERSION + ".json");
+
+        try (Closer closer = Closer.create()) {
+            Files.createDirectories(outputFolder);
+
+            if (!Files.exists(checkPath)) {
+                URL url = new URL(DATA_FILE_DOWNLOAD_URL + app.getPlatform().getDataVersion() + "/" + CLI_DATA_VERSION);
+                ReadableByteChannel readableByteChannel = Channels.newChannel(url.openStream());
+
+                FileOutputStream fileOutputStream = closer.register(new FileOutputStream(checkPath.toFile()));
+                fileOutputStream.getChannel().transferFrom(readableByteChannel, 0, Long.MAX_VALUE);
```

Comment:
I think this makes much more sense:
```suggestion
                URL url = new URL(DATA_FILE_DOWNLOAD_URL + app.getPlatform().getDataVersion() + "/" + CLI_DATA_VERSION);
                try (var stream = url.openStream()) {
                    Files.copy(stream, checkPath);
                }
```